### PR TITLE
[PTQ] Update calibrate.py with the other algorithms support

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Added support of different POT-style algorithms into `calibrate.py`

### Reason for changes

- Extended usage of the `calibrate.py` tool

### Related tickets

- 108993

### CODE REVIEW

File: tests/openvino/tools/calibrate.py [hunk 1]
```diff
@@ -615,18 +635,43 @@ def get_accuracy_checker_config(engine_config):
...
+        nncf_advanced_algo_parameters = MAP_POT_NNCF_ALGORITHMS[pot_algo_name].get("advanced_parameters", {})
+
+        if pot_algo_name in OVERRIDE_OPTIONS_ALGORITHMS:
+            if nncf_algo_name not in override_options:
+                override_options[nncf_algo_name] = defaultdict(dict)
+            parameters = MAP_POT_NNCF_ALGORITHMS[pot_algo_name].get("parameters", {})
+            for p_name, p_value in nncf_advanced_algo_parameters.items():
+                override_options[nncf_algo_name]["advanced_parameters"][p_name] = p_value
+            for p_name, p_value in parameters.items():
+                override_options[nncf_algo_name]["parameters"][p_name] = p_value
```
Comment (andrey-churkin) [inline tests/openvino/tools/calibrate.py:655]: Why not just write like this?
```python
override_options[nncf_algo_name]["advanced_parameters"].update(nncf_advanced_algo_parameters)
override_options[nncf_algo_name]["parameters"].update(parameters)
```

File: tests/openvino/tools/calibrate.py [hunk 2]
```diff
@@ -615,18 +635,43 @@ def get_accuracy_checker_config(engine_config):
...
+            continue
 
-        nncf_algo_name = MAP_POT_NNCF_ALGORITHMS[pot_algo.name]
-        nncf_algorithms.append(
-            {"name": nncf_algo_name, "parameters": map_paramaters(pot_algo.name, nncf_algo_name, pot_algo.params)}
-        )
+        nncf_algo_parameters = map_paramaters(pot_algo_name, nncf_algo_name, pot_algo.params)
+
+        update_advanced_algo_parameters(nncf_algo_parameters, nncf_advanced_algo_parameters)
```
Comment (andrey-churkin) [inline tests/openvino/tools/calibrate.py:660]: I suggest you try to use the `dataclasses.replace()` method

File: tests/openvino/tools/calibrate.py [hunk 3]
```diff
@@ -615,18 +635,43 @@ def get_accuracy_checker_config(engine_config):
...
+        nncf_algorithms[nncf_algo_name] = nncf_algo_parameters
+
+    for override_algo_name, override_values in override_options.items():
+        update_advanced_algo_parameters(nncf_algorithms[override_algo_name], override_values["advanced_parameters"])
+        for p_name, p_value in override_values["parameters"].items():
+            nncf_algorithms[override_algo_name][p_name] = p_value
```
Comment (andrey-churkin) [inline tests/openvino/tools/calibrate.py:666]: `dict.update()` can be used here as well

File: tests/openvino/tools/calibrate.py [hunk 4]
```diff
@@ -51,9 +51,29 @@
...
 TModel = TypeVar("TModel")
 
+OVERRIDE_OPTIONS_ALGORITHMS = ["ActivationChannelAlignment", "FastBiasCorrection", "BiasCorrection"]
+
 MAP_POT_NNCF_ALGORITHMS = {
-    "DefaultQuantization": "quantize",
-    "AccuracyAwareQuantization": "quantize_with_accuracy_control",
+    "ActivationChannelAlignment": {
+        "method": "quantize",
+        "advanced_parameters": {"disable_channel_alignment": False},
```
Comment (andrey-churkin) [inline tests/openvino/tools/calibrate.py:59]: Could you please briefly explain why `"disable_bias_correction"` is not set to `True` here?

File: tests/openvino/tools/calibrate.py (hunk 1 above)
Comment (KodiaqQ) [reply tests/openvino/tools/calibrate.py]: Thanks, updated.

File: tests/openvino/tools/calibrate.py (hunk 2 above)
Comment (KodiaqQ) [reply tests/openvino/tools/calibrate.py]: Very useful, thanks. Updated.

File: tests/openvino/tools/calibrate.py (hunk 3 above)
Comment (KodiaqQ) [reply tests/openvino/tools/calibrate.py]: Also updated.

**Comment by KodiaqQ** [comment]
run pytorch pre-commit tests

**Comment by KodiaqQ** [comment]
@andrey-churkin, @alexsu52, review as possible, please.

**Review by andrey-churkin — APPROVED** [review]

**Pull request merged**